docs(dashboards): one consolidated pass over the Dashboards V2 docs (#3856)
SigNoz/signoz.io: data/docs/userguide/manage-dashboards.mdx

</Admonition>

### Locked view

Selecting the built-in **Locked** view places `locked = true` directly into the visible query box as an editable constraint, rather than applying it as a hidden filter. You can see the constraint, edit it, or combine it with other filters like any search term.

## Row Actions and Role Access on the Dashboards List

Every user sees the per-row actions menu on the dashboards list, but the actions it offers depend on their permission. Read-only actions are available to all roles, and edit actions are available only to users with edit permission (Editor, Author, and Admin roles).

**Read-only actions (all roles, including Viewer):**

- **View**: open the dashboard.
- **Open in New Tab**: open the dashboard in a new browser tab.
- **Copy Link**: copy a direct link to the dashboard.

**Edit actions (Editor, Author, and Admin roles):**

- **Rename**: rename the dashboard.
- **Add Tags** or **Edit Tags**: manage the dashboard's tags. See [Manage Dashboard Tags](#manage-dashboard-tags).
- **Duplicate**: create a copy of the dashboard.

> - **Lock Dashboard** or **Unlock Dashboard**: prevent or re-allow edits.

Reviewer: Lock and unlock look gated on authorship rather than on edit permission, so an Editor or Author who did not create the dashboard may not get this action at all. The Dashboard Options Menu section below states the author/Admin rule. Could you confirm which is right and align the two?

```suggestion
- **Lock Dashboard** or **Unlock Dashboard**: prevent or re-allow edits. Available to the dashboard's author and to Admins, not to everyone with edit permission.
```

### Rename or delete a section

Open a section's **...** (three-dot) menu and select **Rename section** or **Delete section**. Deleting a section also removes the panels it contains.

### Move a panel between sections

1. Open the panel's action menu (the three-dot menu on the panel header).
2. Hover **Move to section**.
3. Choose the target section from the submenu. Select **Dashboard (root)**, the first entry, to move the panel out of any section back to the top level of the dashboard.

The **Move to section** submenu is hidden when there is nowhere to move the panel, for example when the dashboard has no sections and the panel is already at the root.

## Edit a Dashboard as JSON

The dashboard toolbar includes a **JSON** button that opens the JSON editor in a side drawer. Use it to review or hand-edit a dashboard's definition, copy it, or download it, without leaving the page or calling the dashboard API.

A dashboard spec has two parts that must stay in sync:

- **`panels`**: the panels defined on the dashboard.

> - **`layout`**: the sections and positions that place each panel on the grid.

Reviewer: The field looks like `layouts` (plural) in the V2 spec. Since this one is code-formatted, a reader copying it into the JSON editor or an API payload would hit a validation error. Worth verifying?

```suggestion
- **`layouts`**: the sections and positions that place each panel on the grid.
```

3. Edit the JSON. The validation warnings in the drawer footer update as you type.

The drawer toolbar also lets you format the JSON, copy it to your clipboard, download it as a file, and reset your edits.

<Figure
  src="/img/docs/dashboards/json-editor/json-editor-drawer.webp"
  alt="The Dashboard JSON drawer showing the spec, the Format/Copy/Download/Reset toolbar, and a valid-JSON line count"
  caption="The JSON drawer, with the Format, Copy, Download, and Reset controls and a live validity indicator."
/>

<Admonition type="info">
On a locked dashboard, or if you do not have edit access, the editor opens in read-only mode. You can still read, copy, and download the JSON.
</Admonition>

### What the editor exposes

The editor shows only the two editable parts of a dashboard:

- **`tags`**: the dashboard's tags.
- **`spec`**: the dashboard definition, covering panels, layout, variables, and settings.

Server-managed fields (such as `id`, `name`, schema version, image, organization ID, timestamps, and locked status) are hidden from the editor. They are preserved automatically when you save, so nothing is lost by editing only `tags` and `spec`.

> The **Copy** and **Download** controls export exactly what the editor shows, which is the `tags` and `spec` fields only.

Reviewer: `image` appears to be exposed in the editor alongside `tags` and `spec`, with Copy and Download including it. Could you check before we describe it as hidden and server-managed?

```suggestion
Server-managed fields (such as `id`, `name`, schema version, organization ID, timestamps, and locked status) are hidden from the editor. They are preserved automatically when you save, so nothing is lost by editing only `tags`, `spec`, and `image`.

The **Copy** and **Download** controls export exactly what the editor shows, which is the `tags`, `spec`, and `image` fields.
```

<Admonition type="tip">
To move a dashboard between workspaces, download its JSON here and re-import it using [Import JSON](https://signoz.io/docs/dashboards/import-dashboard/).
</Admonition>

### Validation warnings

> The editor shows amber warning banners in the drawer footer when the spec and layout do not match. You can save while a warning is present, but the affected panels will not render as expected until you resolve it.

Reviewer: The two warnings may not behave the same on save. A panel left out of the layout looks saveable, but a layout item pointing at a panel that no longer exists looks like it gets rejected. Could you check which of the two actually blocks a save?

```suggestion
The editor shows amber warning banners in the drawer footer when the spec and layout do not match. A panel left out of the layout saves and simply does not render, but a layout item that points to a panel which no longer exists is rejected on save, so resolve that one first.
```

## Clone a Dashboard

Select **Clone dashboard** from the [Dashboard Options Menu](#dashboard-options-menu) to duplicate a dashboard. SigNoz derives the new dashboard's name from the original:

- The first clone appends `- Copy` to the display name. For example, `My Dashboard` becomes `My Dashboard - Copy`.
- Cloning again increments a counter: `My Dashboard - Copy (2)`, `My Dashboard - Copy (3)`, and so on.
- If the base name is too long to fit the suffix within the [128-character limit](#dashboard-naming-limits), the base name is truncated automatically so the suffix always fits.

Rename the clone at any time from the [Dashboard Options Menu](#dashboard-options-menu) or the configuration drawer.

## Locked and read-only dashboards

A dashboard becomes read-only for one of two reasons, and SigNoz treats them the same way in the UI:

- **The dashboard is locked.** Lock it from the [options menu](#dashboard-options-menu) to prevent accidental edits. Only the dashboard's author and Admins can lock or unlock.
- **You do not have edit permission.** Viewers have read-only access to every dashboard.

> In read-only mode, editing controls stay visible but disabled rather than being hidden, so the dashboard looks the same to everyone. Hovering a disabled control explains why, with either `This dashboard is locked` or `You don't have permission to edit this dashboard`.

Reviewer: These two read-only modes may not look the same in the UI: on a locked dashboard the controls seem to render disabled, while a user without edit permission seems to have them hidden. The **Clone dashboard** bullet below may also be unavailable to Viewers for the same reason. Could you check both before we describe them as identical?

```suggestion
On a locked dashboard, editing controls stay visible but disabled, and hovering one explains why with `This dashboard is locked`. Without edit permission the controls are hidden instead, so a Viewer never sees them.
```
